### PR TITLE
Admin user and event search

### PR DESCRIPTION
**Pull request by Wumbo12347**
fixed showing all events when the page loads for both admin user and event search.

added cascading deletes for users and events

### CODE REVIEW

**Comment by Sherali5543** [comment]
please add some tests

File: napkinapp/app/src/main/java/com/example/napkinapp/fragments/admineventsearch/AdminListEventsFragment.java [hunk 1]
```diff
@@ -123,7 +124,6 @@ public void onFailure(Exception e) {
                 Log.e("RegisteredEventsFragment", "Error loading events: " + e.getMessage(), e);
             }
         }, Event.class);
-
     }
 
     private void
```
Comment (antoszko) [inline napkinapp/app/src/main/java/com/example/napkinapp/fragments/admineventsearch/AdminListEventsFragment.java:129]: formatting?

**Comment by antoszko** [comment]
It doesn't actually do a deep delete for delete user nor delete event.
For delete event, you need to loop through all the waitlist, chosen, and registered lists, go into each user stored there, and remove that eventId from the user's respective list.

For delete user, you  need to loop through all their waitlist, chosen, and registered lists, go into each event in that list, and remove that userId from the event's respective list,
Additionally for delete user, you need loop through each of the user's events and call the deleteEvent function. ^

File: napkinapp/app/src/main/java/com/example/napkinapp/fragments/adminusersearch/AdminListUsersFragment.java [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+    private DB_Client db;
+    public AdminListUsersFragment() {
+        // Required null constructor
+    }
+
+    AdminUserArrayAdapter.UserListCustomizer customizer = button -> {
+        button.setText("Remove");
+        button.setOnClickListener(v -> {
+            User user = (User) v.getTag();
+            Log.i("Button", String.format("List Events: Clicked on event %s\n", user.getName()));
```
Comment (antoszko) [inline napkinapp/app/src/main/java/com/example/napkinapp/fragments/adminusersearch/AdminListUsersFragment.java:51]: comment should say user not event

File: napkinapp/app/src/main/java/com/example/napkinapp/fragments/adminusersearch/AdminListUsersFragment.java [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+                }
+                @Override
+                public void onFailure(Exception e) {
+                    Toast.makeText(mContext, "Failed to get event: " + e.getMessage(), Toast.LENGTH_SHORT).show();
+                    Log.e("RegisteredEventsFragment", "Error getting event", e);
+                }
+            }, Event.class);
+        }
+
+        // remove events which the deleted user has created
```
Comment (antoszko) [inline napkinapp/app/src/main/java/com/example/napkinapp/fragments/adminusersearch/AdminListUsersFragment.java:302]: isn't this already handled in deleteUserFromCancelledListsANDORGANIZEDEVENTS??

Comment (antoszko) [reply napkinapp/app/src/main/java/com/example/napkinapp/fragments/adminusersearch/AdminListUsersFragment.java:302]: Also this is shallow delete so would need to fix that anyways. but I beleive you already do this in tha aforementioned function

File: napkinapp/app/src/main/java/com/example/napkinapp/fragments/adminusersearch/AdminListUsersFragment.java [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+            public void onSuccess(Void aVoid) {
+                Toast.makeText(mContext, "User deleted successfully", Toast.LENGTH_SHORT).show();
+                users.remove(user); // Remove the event from the local list
+                userArrayAdapter.notifyDataSetChanged(); // Update the adapter
+                Log.d("RegisteredEventsFragment", "User deleted from Firestore and list updated.");
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                Toast.makeText(mContext, "Failed to delete event: " + e.getMessage(), Toast.LENGTH_SHORT).show();
```
Comment (antoszko) [inline napkinapp/app/src/main/java/com/example/napkinapp/fragments/adminusersearch/AdminListUsersFragment.java:325]: toast message

File: napkinapp/app/src/main/java/com/example/napkinapp/fragments/adminusersearch/AdminListUsersFragment.java [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+        button.setText("Remove");
+        button.setOnClickListener(v -> {
+            User user = (User) v.getTag();
+            Log.i("Button", String.format("List Events: Clicked on event %s\n", user.getName()));
+
+
+            // delete the user from the event waitlist
+
+
+            Toast.makeText(mContext, "Delete User", Toast.LENGTH_SHORT).show();
```
Comment (antoszko) [inline napkinapp/app/src/main/java/com/example/napkinapp/fragments/adminusersearch/AdminListUsersFragment.java:57]: remove this toast, deleteUser() already does toasts

**Review by antoszko — CHANGES_REQUESTED** [review]

**Review by antoszko — APPROVED** [review]

**Pull request merged**